indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -4381,6 +4374,39 @@ public ConcurrentHashMap<PartitionIdType, SequenceOffsetType> getPartitionOffset
     return partitionOffsets;
   }
 
+  /**
+   * Calculates the average poll-idle-ratio metric across all active tasks.
+   * This metric indicates how much time the consumer spends idle waiting for data.
+   *
+   * @return the average poll-idle-ratio across all tasks, or 1 (full idle) if no tasks or metrics are available
+   */
+  public double getPollIdleRatioMetric()
+  {
+    Map<String, Map<String, Object>> taskMetrics = getStats(true);
+    if (taskMetrics.isEmpty()) {
+      return 1.;
```

Comment:
Returning 1 (full idle) here would cause the auto-scaler to think that the tasks are doing nothing and cause a scale-down, when in fact the tasks failed to return the metrics and may be in a bad state. Scaling down might further worsen the problem.

Should we return 0 instead?

Reply:
Scaling up might be an overkill in that scenario - nothing may happen and instead we will waste a resources.
**0.5** looks optimal for me (during the implementation i thought between 0.5 and 1).

Reply:
We can even return -1 to denote that we do not have metrics available, and just skip scaling rather than make a bad decision.